src/models/network/NetworkLibController.js

```diff
@@ -27,11 +25,24 @@ export default class NetworkLibController extends NetworkBaseController {
     this.package = await Package.fetch(address, this.txParams);
   }
 
+  async push(reupload = false) {
+    const requestedVersion = this.packageData.version;
+    const currentVersion = this.networkPackage.version;
+    await super.push(reupload)
+    if (requestedVersion !== currentVersion) this.networkPackage.frozen = false
```

Comment:
Instead of overriding `push`, include `this.networkPackage.frozen = false` in `newVersion` directly, since it is only called when `requestedVersion !== currentVersion`.